### PR TITLE
tests/ci: add tag/clean aws jenkins jobs as a code.

### PR DESCRIPTION
**Pull request by cpanato**
Add the Clean and Tag AWS jobs as a jenkins script.

After the Merge of this PR, I will delete the existing tag/clean-aws jobs and we will start to use the new ones.

Every time we need to bump the grafiti image we need to change the code in the repo to update the grafiti docker image.

Relies on #1239

### CODE REVIEW

**Comment by cpanato** [comment]
cc @mxinden @estroz @squat

**Comment by estroz** [comment]
@cpanato moved jobs into separate scripts. Each job now leverages maintenance scripts to be merged in #1239

**Comment by cpanato** [comment]
@estroz LGVTM Thanks for the update :D I made a comment in https://github.com/coreos/tectonic-installer/pull/1239#issuecomment-319958207 to make sure we don't need extra bindings. If dont need to login into quay I'm fine with this PR

**Comment by cpanato** [comment]
@mxinden @estroz I will combine in one script

**Comment by cpanato** [comment]
@mxinden @estroz done

**Comment by mxinden** [comment]
@cpanato What do you think of https://github.com/coreos/tectonic-installer/pull/1564#pullrequestreview-54345850?

**Comment by mxinden** [comment]
@cpanato And can you rebase the PR? Got out of sync since #1573 is merged.

**Comment by cpanato** [comment]
@mxinden Now I got what you wrote about join both scripts in just one. I thought to join the groovy script. My bad :( I agree I will update that.

**Comment by cpanato** [comment]
@mxinden update the code to use just one script. Tested in my sample and works. Now we need to wait https://github.com/coreos/tectonic-installer/pull/1239 to be merged and then we can merge this one

please let me know what you think

File: tests/jenkins-jobs/maintenance/grafiti_tag_clean_aws.groovy [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    --force \\
+    \\\${DATE_OVERRIDE_FLAG}
+done
+\"\"\".stripIndent()
+      }
+    }
+  }
+    catch (err) {
+      currentBuild.result = "FAILURE"
+      slackSend channel: '#tectonic-installer-ci', color: 'danger', message: 'Jenkins Maintenance: ' + env.JOB_NAME + ' Please go to ' + env.BUILD_URL + ' and to check the build', tokenCredentialId: 'tectonic-slack-token'
```
Comment (mxinden) [inline tests/jenkins-jobs/maintenance/grafiti_tag_clean_aws.groovy]: s/and to check the build/and check the build/

File: tests/jenkins-jobs/maintenance/grafiti_tag_clean_aws.groovy [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+H H/6 * * * % TAG_CLEAN=clean-aws
+    """.stripIndent())
+    }
+  }
+
+  definition {
+    cps {
+      sandbox(true)
+      script("""
+node('master') {
```
Comment (mxinden) [inline tests/jenkins-jobs/maintenance/grafiti_tag_clean_aws.groovy]: Why is this written as a string and then parsed?

Comment (cpanato) [inline tests/jenkins-jobs/maintenance/grafiti_tag_clean_aws.groovy]: I will split this in another file and load here. Will be better for reading

Comment (mxinden) [inline tests/jenkins-jobs/maintenance/grafiti_tag_clean_aws.groovy]: I am just confused why we are using a workflow definition here?

File: tests/jenkins-jobs/maintenance/grafiti_tag_clean_aws.groovy (hunk 1 above)
Comment (cpanato) [inline tests/jenkins-jobs/maintenance/grafiti_tag_clean_aws.groovy]: done

File: tests/jenkins-jobs/maintenance/grafiti_tag_clean_aws.groovy (hunk 2 above)
Comment (cpanato) [inline tests/jenkins-jobs/maintenance/grafiti_tag_clean_aws.groovy]: because It is a little bit better when using the parametrized cron

Comment (cpanato) [inline tests/jenkins-jobs/maintenance/grafiti_tag_clean_aws.groovy]: I will keep this for now, I need more experiments to split that.

if you want to see the job please look into this https://jenkins-tectonic-installer.prod.coreos.systems/job/maintenance/job/tag_clean_aws_resources_grafiti/configure

after let me know then I will delete that. I also disable the job to not run this is just for your validation

Comment (mxinden) [inline tests/jenkins-jobs/maintenance/grafiti_tag_clean_aws.groovy]: @cpanato Why don't we use `script` like we do in the main [Jenkinsfile](https://github.com/coreos/tectonic-installer/blob/master/Jenkinsfile#L145)?

File: tests/jenkins-jobs/maintenance/tag_clean_aws_grafiti_job.groovy [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      parameterizedSpecification("""H H/6 * * * % TAG_CLEAN=tag-aws
+H H/6 * * * % TAG_CLEAN=clean-aws
+    """.stripIndent())
+    }
+  }
+
+  definition {
+    cps {
+      sandbox(true)
+      script("""
```
Comment (sym3tri) [inline tests/jenkins-jobs/maintenance/tag_clean_aws_grafiti_job.groovy]: Is it possible to load this from a file or something so we can avoid all the escaping?

Comment (cpanato) [inline tests/jenkins-jobs/maintenance/tag_clean_aws_grafiti_job.groovy]: yes! will do that

File: tests/jenkins-jobs/maintenance/grafiti_tag_clean_aws.groovy (hunk 2 above)
Comment (cpanato) [inline tests/jenkins-jobs/maintenance/grafiti_tag_clean_aws.groovy]: will change that

**Comment by cpanato** [comment]
@mxinden @sym3tri split to avoid too many escaping.

**Comment by mxinden** [comment]
@cpanato Let's talk about this quickly in our meeting at 3 before merging.

**Comment by cpanato** [comment]
@mxinden I was able to achieve the same results using a normal job :) please take a look again and let me know your comments

**Comment by cpanato** [comment]
👍

**Comment by mxinden** [comment]
@cpanato Feel free to merge if there is nothing else needed. #1239 is merged.

**Comment by cpanato** [comment]
@mxinden rebased please review again 

thanks!

**Pull request merged**